capsules/extra/src/usb_hid_driver.rs

```diff
@@ -112,24 +82,17 @@ impl<'a, U: usb_hid::UsbHid<'a, [u8; 64]>> usb_hid::Client<'a, [u8; 64]> for Usb
         _endpoint: usize,
     ) {
         self.processid.map(|id| {
-            self.app
-                .enter(id, |app, kernel_data| {
-                    let _ = kernel_data
-                        .get_readwrite_processbuffer(rw_allow::RECV)
-                        .and_then(|recv| {
-                            recv.mut_enter(|dest| {
-                                dest.copy_from_slice(buffer);
-                            })
-                        });
-
-                    kernel_data.schedule_upcall(0, (0, 0, 0)).ok();
-                    app.can_receive.set(false);
-                })
-                .map_err(|err| {
-                    if err == kernel::process::Error::NoSuchApp
-                        || err == kernel::process::Error::InactiveApp
-                    {}
-                })
+            let _ = self.app.enter(id, |_app, kernel_data| {
```

Comment:
This is fine here, as it's just adapting to a pattern we are using elsewhere in the kernel already, but I wonder if we should consider adding a `enter_if_alive()` method or similar, rather than propagating a pattern of ignoring (meaningful) `Result`s in the kernel.

Reply:
But then we have to ensure that `enter_if_alive()` is only called when appropriate (i.e. in callbacks where there is nothing useful to do with the error), and not when the error could have been used.